src/components/demo-step-0.js

```diff
@@ -0,0 +1,107 @@
+import React, { Component } from 'react'
+import { render } from 'react-dom'
+
+import contractService from '../services/contract-service'
+import ipfsService from '../services/ipfs-service'
+import bs58 from 'bs58'
+
+class DemoStep0 extends Component {
+
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      listingsResults: [],
+      contractListingsCount: -1
+    }
+
+    let that = this
+
+    // Test getting listings from chain
+    setTimeout(function() {
+      // TODO: Remove hacky 2s delay and correctly determine when contractService
+      // is ready
+      contractService.getAllListings().then((allContractResults) => {
+        var resultIndex;
+        console.log("Got this many results:" + allContractResults.length)
+        console.log(allContractResults)
+        that.setState({contractListingsCount: allContractResults.length})
+        for (resultIndex in allContractResults) {
+          (function (contractResult) {
+            const hashStr = contractResult.ipfsHash
+            ipfsService.getListing(hashStr)
+            .then((listingJson) => {
+              const listingData = {
+                'contract': contractResult,
+                'ipfs': JSON.parse(listingJson).data
+              }
+              console.log(listingData)
+              // Append our new result to state. For now we don't care about ordering.
+              that.setState({
+                listingsResults: that.state.listingsResults.concat(listingData)
+              })
+            })
+            .catch((error) => {
+              alert(error)
```

Comment:
Fine for now. Future request for the future is some nicer dialog or toast error message. :)

Reply:
![toast](https://user-images.githubusercontent.com/673455/31571400-0fdf6450-b04f-11e7-9716-6ae287a3a953.jpg)